core/src/main/scala/akka/kafka/ConsumerSettings.scala

```diff
@@ -20,21 +22,50 @@ import scala.collection.JavaConverters._
 import scala.compat.java8.OptionConverters._
 import scala.concurrent.duration._
 
-sealed trait Subscription
-sealed trait ManualSubscription extends Subscription
-sealed trait AutoSubscription extends Subscription
+sealed trait Subscription {
+  def rebalanceListenerCallbacks: Option[ListenerCallbacks]
+  /** Sets callbacks ti be invoked when the consumer related to this subscription gets rebalancing events */
+  def withRebalanceListenerCallbacks(onAssign: Set[TopicPartition] => Unit, onRevoke: Set[TopicPartition] => Unit): Subscription
+  /** Java API: Sets callbacks ti be invoked when the consumer related to this subscription gets rebalancing events */
+  final def withRebalanceListenerCallbacksJavadsl(onAssign: Procedure[java.util.Set[TopicPartition]], onRevoke: Procedure[java.util.Set[TopicPartition]]): Subscription =
```

Comment:
the api naming is a mess here and there, we should take another round of cleaning up the API. Created https://github.com/akka/reactive-kafka/issues/424

Reply:
Yeah, cleanup would be nice... I have not found a nice way without breaking API though